mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -28,144 +32,524 @@ impl IndexedDbNftStorage {
+            .await?;
+        let last_scanned_block = LastScannedBlockTable {
+            chain: chain.to_string(),
+            last_scanned_block: BeBigUint::from(scanned_block),
+        };
+        last_scanned_block_table
+            .replace_item_by_unique_index("chain", chain.to_string(), &last_scanned_block)
+            .await?;
+        Ok(())
     }
 
-    async fn update_nft_amount_and_block_number(&self, _chain: &Chain, _nft: Nft) -> MmResult<(), Self::Error> {
-        todo!()
+    async fn update_nft_amount_and_block_number(&self, chain: &Chain, nft: Nft) -> MmResult<(), Self::Error> {
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let nft_table = db_transaction.table::<NftListTable>().await?;
+        let last_scanned_block_table = db_transaction.table::<LastScannedBlockTable>().await?;
+
+        let index_keys = MultiIndex::new(NftListTable::CHAIN_TOKEN_ADD_TOKEN_ID_INDEX)
+            .with_value(chain.to_string())?
+            .with_value(&nft.token_address)?
+            .with_value(nft.token_id.to_string())?;
+
+        let nft_item = NftListTable::from_nft(&nft)?;
+        nft_table
+            .replace_item_by_unique_multi_index(index_keys, &nft_item)
+            .await?;
+        let last_scanned_block = LastScannedBlockTable {
+            chain: chain.to_string(),
+            last_scanned_block: BeBigUint::from(nft.block_number),
+        };
+        last_scanned_block_table
+            .replace_item_by_unique_index("chain", chain.to_string(), &last_scanned_block)
+            .await?;
+        Ok(())
     }
 }
 
 #[async_trait]
 impl NftTxHistoryStorageOps for IndexedDbNftStorage {
     type Error = WasmNftCacheError;
 
-    async fn init(&self, _chain: &Chain) -> MmResult<(), Self::Error> { todo!() }
+    async fn init(&self, _chain: &Chain) -> MmResult<(), Self::Error> { Ok(()) }
 
-    async fn is_initialized(&self, _chain: &Chain) -> MmResult<bool, Self::Error> { todo!() }
+    async fn is_initialized(&self, _chain: &Chain) -> MmResult<bool, Self::Error> { Ok(true) }
 
     async fn get_tx_history(
         &self,
-        _chains: Vec<Chain>,
-        _max: bool,
-        _limit: usize,
-        _page_number: Option<NonZeroUsize>,
-        _filters: Option<NftTxHistoryFilters>,
+        chains: Vec<Chain>,
+        max: bool,
+        limit: usize,
+        page_number: Option<NonZeroUsize>,
+        filters: Option<NftTxHistoryFilters>,
     ) -> MmResult<NftsTransferHistoryList, Self::Error> {
-        todo!()
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let table = db_transaction.table::<NftTxHistoryTable>().await?;
+        let mut txs = Vec::new();
+        for chain in chains {
+            let tx_tables = table
+                .get_items("chain", chain.to_string())
```

Comment:
The `get_items` method is used to fetch all items from the `NftTxHistoryTable` that match a given chain. While this is fine for smaller datasets, it might become inefficient if the table grows large, as it retrieves all items regardless of how many are needed. While we don't need to do this part at this stage but It would be good to have a `get_items_with_limit` in the future that uses [`get_all_with_key_and_limit`](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.IdbObjectStore.html#method.get_all_with_key_and_limit) instead of [`get_all_with_key`](https://github.com/KomodoPlatform/atomicDEX-API/blob/211492e586d301c7ba0da56585bf6b759782fd23/mm2src/mm2_db/src/indexed_db/drivers/object_store.rs#L61) we might even go further and try to filter and sorting before retrieving from db so we won't need `take_txs_according_to_filters` or `take_txs_according_to_paging_opts`
I will open an issue for this since it's needed in multiple places throughout the code. @laruh As I said, no need to fix this in the current PR, I was just noting it :)